pkg/agent/cniserver/pod_configuration_windows.go

```diff
@@ -134,3 +135,169 @@ func (pc *podConfigurator) reconcileMissingPods(ifConfigs []*interfacestore.Inte
 		}
 	}
 }
+
+type unReadyPodInfo struct {
+	podName      string
+	podNamespace string
+	annotated    bool
+	createTime   time.Time
+}
+
+type podIfaceMonitor struct {
+	kubeClient        clientset.Interface
+	ifaceStore        interfacestore.InterfaceStore
+	ofClient          openflow.Client
+	podUpdateNotifier channel.Notifier
+
+	// unReadyInterfaces is a map to store the OVS ports which is waiting for the PortStatus from OpenFlow switch.
+	// The key in the map is the OVS port name, and its value is unReadyPodInfo.
+	// It is used only on Windows now.
+	unReadyInterfaces sync.Map
+	statusCh          chan *openflow15.PortStatus
+}
+
+func newPodInterfaceMonitor(kubeClient clientset.Interface,
+	ofClient openflow.Client,
+	ifaceStore interfacestore.InterfaceStore,
+	podUpdateNotifier channel.Notifier,
+) *podIfaceMonitor {
+	statusCh := make(chan *openflow15.PortStatus)
+	ofClient.SubscribeOFPortStatusMessage(statusCh)
+	return &podIfaceMonitor{
+		kubeClient:        kubeClient,
+		ofClient:          ofClient,
+		ifaceStore:        ifaceStore,
+		podUpdateNotifier: podUpdateNotifier,
+		unReadyInterfaces: sync.Map{},
+		statusCh:          statusCh,
+	}
+}
+
+func (m *podIfaceMonitor) monitorUnReadyInterface(stopCh <-chan struct{}) {
+	klog.Info("Started the monitor to wait for new OpenFlow ports")
+	go func() {
+		for {
+			select {
+			case <-stopCh:
+				return
+			case status := <-m.statusCh:
+				klog.V(2).InfoS("Received PortStatus message", "message", status)
+				// Update Pod OpenFlow entries only after the OpenFlow port state is live.
+				if status.Desc.State == openflow15.PS_LIVE {
+					m.updateUnReadyPod(status)
+				}
+			case <-time.Tick(time.Second * 5):
+				m.checkUnReadyPods()
+			}
+		}
+	}()
+}
+
+func (m *podIfaceMonitor) updatePodFlows(ifName string, ofPort int32) error {
```

Comment:
Have you considered making this configurable: when instantiating the podIfaceMonitor, a function is provided which will be called when a Pod becomes "ready". This way, this object (`podIfaceMonitor`) doesn't need to be aware of things like `ifaceStore`, `ofClient` and `podUpdateNotifier`. That logic can be provided by the consumer of the `podIfaceMonitor`.

Reply:
I feel it is not necessary to make it configurable for now unless we expect to leverage the OpenFlow PortStatus message to perform some other actions rather than install OpenFlow entries.